x-pack/plugins/security_solution/common/detection_engine/schemas/types/threat_mapping.ts

```diff
@@ -19,29 +19,41 @@ export type ThreatFilters = t.TypeOf<typeof threat_filters>;
 export const threatFiltersOrUndefined = t.union([threat_filters, t.undefined]);
 export type ThreatFiltersOrUndefined = t.TypeOf<typeof threatFiltersOrUndefined>;
 
-export const threatMappingEntries = t.array(
-  t.exact(
-    t.type({
-      field: NonEmptyString,
-      type: t.keyof({ mapping: null }),
-      value: NonEmptyString,
-    })
-  )
+export const threatMapEntry = t.exact(
+  t.type({
+    field: NonEmptyString,
+    type: t.keyof({ mapping: null }),
+    value: NonEmptyString,
+  })
 );
+
+export type ThreatMapEntry = t.TypeOf<typeof threatMapEntry>;
+
+export const threatMappingEntries = t.array(threatMapEntry);
 export type ThreatMappingEntries = t.TypeOf<typeof threatMappingEntries>;
 
-export const threat_mapping = t.array(
-  t.exact(
-    t.type({
-      entries: threatMappingEntries,
-    })
-  )
+export const threatMap = t.exact(
+  t.type({
+    entries: threatMappingEntries,
+  })
 );
+export type ThreatMap = t.TypeOf<typeof threatMap>;
+
+export const threat_mapping = t.array(threatMap);
 export type ThreatMapping = t.TypeOf<typeof threat_mapping>;
 
 export const threatMappingOrUndefined = t.union([threat_mapping, t.undefined]);
 export type ThreatMappingOrUndefined = t.TypeOf<typeof threatMappingOrUndefined>;
 
-export const threat_index = t.string;
+export const threat_index = t.array(t.string);
+export type ThreatIndex = t.TypeOf<typeof threat_index>;
 export const threatIndexOrUndefined = t.union([threat_index, t.undefined]);
 export type ThreatIndexOrUndefined = t.TypeOf<typeof threatIndexOrUndefined>;
+
+export const threat_language = t.union([
+  t.keyof({ eql: null, kuery: null, lucene: null }),
```

Comment:
Could this just be a union of `Language` and `undefined`?

Reply:
Of course, of course. The less types the better. Thank you for the feedback (as always), I will fix it here.